crates/interpreter/src/side_table.rs

```
@@ -137,10 +150,10 @@ impl BranchTableEntry {
     }
 
     pub fn view(self) -> BranchTableEntryView {
-        let pop_val_counts = parse_u16(&self.0, 4);
+        let pop_val_counts = parse_u16::<Use>(&self.0, 4).unwrap();
         BranchTableEntryView {
-            delta_ip: (parse_u16(&self.0, 0) as i16) as i32,
-            delta_stp: (parse_u16(&self.0, 2) as i16) as i32,
+            delta_ip: (parse_u16::<Use>(&self.0, 0).unwrap() as i16) as i32,
+            delta_stp: (parse_u16::<Use>(&self.0, 2).unwrap() as i16) as i32,
```

Same here, we want this function to be parametric over the mode. We use it in `patch_branch()` during verification. And actually, we should not panic when indexing with `[source.branch_table]` there. We should also use `get()` (unless we have a test somewhere else).

---

Where is the "indexing" in this function?

---

There's no "indexing" in this function. The indexing I mentioned was where it is called in `valid.rs`. That can be done in the next PR. I hope I'll be able to find out all those issues in the final review from `dev/fast-interp` to `main`. There are actually many places where we would panic instead of returning an error on user errors.